### PR TITLE
Feat/support animation

### LINKED ISSUE #63: Request: support code block animation
A great extension, thanks for that!

Would it be possible to support code animation?
An example in RevealJS HTML:
```html
<section data-auto-animate>
     <h2 data-id="code-title">Even Prettier Animations</h2>
     <pre data-id="code-animation">
       <code class="hljs" data-trim data-line-numbers="1|3,6-7|8">
         1 hello
         2 another line
         3 more line
         4 another line
         5 line
         6 yes a line
         7 line
         8 last line
       </code>
     </pre>
   </section>
```

Maybe support for the `data-auto-animate` attribute in an `revealjs_section` directive, and support for the `code-animation` and `data-line-numbers` attributes in the `code-block` directive ?

See also https://revealjs.com/auto-animate/ and https://revealjs.com/code/

Thanks for taking it into consideration

### PR DESCRIPTION
**Pull request by PeterMosmans**
This resolves #63 :)

Note that this is a very rudimentary implementation of the animation attributes, as well as source code highlight animations. I haven't changed/added any tests.

Feel free to modify/append/...

Thanks,

Peter

### CODE REVIEW

File: sphinx_revealjs/writers.py [hunk 1]
```diff
@@ -86,7 +86,19 @@ def visit_literal_block(self, node: literal_block):
...
         lang = node["language"]
-        self.body.append(f'<pre><code data-trim data-noescape class="{lang}">\n')
+        self.body.append("<pre")
+        # add section id as data-id
+        if isinstance(node.parent, section) and len(node.parent["ids"]):
+            self.body.append(f" data-id=\"{node.parent['ids'][0]}\"")
+        self.body.append(f'><code data-trim data-noescape class="{lang}"')
+        # use the emphasize-lines directive to create line for line animations
+        if "highlight_args" in node and "hl_lines" in node["highlight_args"]:
+            self.body.append(f"data-line-numbers=\"{'|'.join(map(str,node['highlight_args']['hl_lines']))}\"")
```
Comment (attakei) [inline sphinx_revealjs/writers.py:97]: `emphasize-lines` option are converted from syntax of option to list of numbers in Sphinx
(ex: `1-3,5` is converted to `[1,2,3,5]` ).

This implementation highlight single line only in order (`[1,2,3,5]`=>`1|2|3|5`).
It does not match expection by user that want to higilight multi-lines.

**Review by attakei — APPROVED** [review]
`emphasize-lines` is information for only line highlighting, and not for order of highlighting. And internal process of sphinx does not keep syntax of source. 

I think that there are three approaches for this problem.

- Define yet another code-block directive that has attribute for code-animation
- Override code-block directive for having attribute for code-animation
- Not implement (revert blocks for code-animation)

Do you have other idea?

File: sphinx_revealjs/writers.py (hunk 1 above)
Comment (PeterMosmans) [reply sphinx_revealjs/writers.py:97]: Thanks for your feedback. Yeah, the solution is far from ideal, it's a bit of hackery.
Personally I think that overriding the code-block directive is the cleanest solution, adding the 'extra' attributes like `line-numbers` and `data-id`. But this needs some work... :flushed:

**Comment by attakei** [comment]
I'm trying implement `reavealjs-code-block` inherit from `code-block` to manage extra attributes for reveal.js on [branch 'feat/support_animation--with-extra-code-block' ](https://github.com/PeterMosmans/sphinx-revealjs/compare/feat/support_animation...attakei:feat/support_animation--with-extra-code-block)

**Comment by attakei** [comment]
@PeterMosmans I committed these to your branch.

- Implement to animate hilighting for code blocks
- Updates demo for animate 

Please try this branch wheather it can work as your feature suggestions.

**Comment by PeterMosmans** [comment]
Thanks @attakei , much appreciated! That looks awesome. I'm going to take it for a spin and let you know.

**Comment by PeterMosmans** [comment]
Hi @attakei - all looks well. I'm looking forward to finding these added features in the official branch.

Thanks a lot for your work!

**Pull request merged**